src/compiler/emitter.ts

```diff
@@ -2228,6 +2233,15 @@ namespace ts {
 
                     // Write the delimiter if this is not the first node.
                     if (previousSibling) {
+                        // i.e
+                        //      function commentedParameters(
+                        //          /* Parameter a */
+                        //          a
+                        //          /* End of parameter a */ -> this comment isn't considered to be trailing comment of parameter "a" due to newline
+                        //          ,
+                        if (emitLeadingCommentsOfPosition && delimiter && previousSibling.end !== parentNode.end) {

```

Comment:
There shouldn't be a need to test for the existence of `emitLeadingCommentsOfPosition`.